app/src/main/java/io/gnosis/safe/ui/safe/empty/NoSafeViewModel.kt

```diff
@@ -0,0 +1,40 @@
+package io.gnosis.safe.ui.safe.empty
+
+import io.gnosis.data.models.Safe
+import io.gnosis.safe.di.Repositories
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.collect
+
+class NoSafeViewModel @Inject constructor(
+    repositories: Repositories,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<SafesState>(appDispatchers) {
+
+    private val safeRepository = repositories.safeRepository()
+
+    override fun initialState() = SafesState(null, 0, ViewAction.Loading(true))
+
+    init {
+        safeLaunch {
+            safeRepository.activeSafeFlow().collect { safe ->
+                val safeNum = safeRepository.getSafes().count()
+                updateState {
+                    SafesState(
+                        safe, safeNum, if (safe == null) ViewAction.None else ViewAction.NavigateTo(
```

Comment:
Not important now, but I think it's more readable to extract this into a `val viewAction` since `updateState { }` is a callback after all